### PR TITLE
dry-types: add some basic examples to intro page

### PR DESCRIPTION
**Pull request by georgemillo**
Add some quick and dirty code examples so that someone coming newly to the gem can get an overview of what the gem is about, and its basic usage.

### CODE REVIEW

File: source/gems/dry-types/index.html.md [hunk 1]
```diff
@@ -16,7 +16,80 @@ sections:
...
+Types::Bool[false]
+# => false
+Types::Bool["not a bool"]
+# Dry::Types::ConstraintError: "not a bool" violates constraints
+```
+
+See [Built-In Types](/gems/dry-types/built-in-types/) for a full list of types.
+
+You can use dry-types with [dry-struct](http://dry-rb.org/gems/dry-struct)
+to create a class whose attributes have clearly-defined, coercible types:
```
Comment (solnic) [inline source/gems/dry-types/index.html.md]: Maybe just "to create a class with typed attributes"? They don't have to be coercible, or even strict, it depends on the use cases.

Comment (georgemillo) [inline source/gems/dry-types/index.html.md]: fair point. Fixed ;)

**Comment by georgemillo** [comment]
Just rewrote this for clarity with more info based on our conversation [here](https://github.com/dry-rb/dry-types/issues/163), and added a new example about `meta`.

**Comment by flash-gordon** [comment]
@solnic is it good to go? This is the last one :)

**Comment by solnic** [comment]
Sorry, I wasn't watching this project and missed this one

**Pull request merged**